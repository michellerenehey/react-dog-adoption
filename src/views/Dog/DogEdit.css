.message {
  color: green;
  font-weight: bold;
  font-size: larger;
}
