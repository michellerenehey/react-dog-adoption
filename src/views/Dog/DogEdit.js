import { useState, useEffect } from 'react';
import { fetchDogById } from '../../services/dogs';

export default function DogEdit(props) {
  const id = props.match.params.id;
  const [name, setName] = useState('');
  const [breed, setBreed] = useState('');
  const [image, setImage] = useState('');
  const [bio, setBio] = useState('');
  console.log(id);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogById(id);
      setName(data.name);
      setBreed(data.breed);
      setImage(data.image);
      setBio(data.bio);
      console.log(data);
    };
    fetchData();
  }, [id]);
  return <div>i am the dog edit component</div>;
}
