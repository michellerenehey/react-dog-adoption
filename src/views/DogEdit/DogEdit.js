import { useState, useEffect } from 'react';
import { fetchDogById, updateDog } from '../../services/dogs';
import { useHistory } from 'react-router-dom';
import DogForm from '../../components/DogForm/DogForm';
import './DogEdit.css';

export default function DogEdit(props) {
  const id = props.match.params.id;
  const [name, setName] = useState('');
  const [breed, setBreed] = useState('');
  const [image, setImage] = useState('');
  const [bio, setBio] = useState('');
  const [message, setMessage] = useState(null);
  const [buttonClass, setButtonClass] = useState('');

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogById(id);
      setName(data.name);
      setBreed(data.breed);
      setImage(data.image);
      setBio(data.bio);
    };
    fetchData();
  }, [id]);

  const history = useHistory();

  const handleSubmit = async (e) => {
    try {
      e.preventDefault();
      await updateDog(id, name, breed, image, bio);
      setMessage('Success! Redirecting...');
      setButtonClass('hidden');
      setTimeout(() => {
        history.goBack();
      }, 2500);
    } catch {
      setMessage('Oh no! Something went wrong!');
    }
  };

  return (
    <div>
      {message && <p className="message">{message}</p>}
      <DogForm
        name={name}
        setName={setName}
        breed={breed}
        setBreed={setBreed}
        image={image}
        setImage={setImage}
        bio={bio}
        setBio={setBio}
        handleSubmit={handleSubmit}
        buttonClass={buttonClass}
      />
    </div>
  );
}
