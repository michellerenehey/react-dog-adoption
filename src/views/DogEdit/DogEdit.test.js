import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import DogEdit from './DogEdit';

test('should render the DogEdit component', async () => {
  const { container } = render(
    <MemoryRouter>
      <DogEdit match={{ params: { id: 3 } }} />
    </MemoryRouter>
  );
  await screen.findByDisplayValue('Poppy');
  expect(container).toMatchSnapshot();
});
