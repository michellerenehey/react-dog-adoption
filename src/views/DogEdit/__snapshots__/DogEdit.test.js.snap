// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the DogEdit component 1`] = `
<div>
  <div>
    <div
      class="dog-form-div"
    >
      <form
        class="dog-form"
      >
        <label
          class="form-detail"
        >
          Dog Name:
        </label>
        <input
          class="form-detail"
          placeholder="enter name"
          required=""
          type="text"
          value="Poppy"
        />
        <label>
          Dog Breed:
        </label>
        <input
          placeholder="enter breed"
          type="text"
          value="Border collie mix"
        />
        <label>
          Image Link:
        </label>
        <input
          placeholder="enter image url"
          type="text"
          value="https://placedog.net/1000/563?id=125"
        />
        <label>
          Dog Bio:
        </label>
        <textarea
          class="dog-bio"
          placeholder="enter bio"
        >
          What a cutie. Poppy is gentle and calm. She loves babies, dandelions, and long walks. Her ears flop when she runs, and she'll snuggle with you all day long. 
        </textarea>
        <button
          class=""
        >
          Save
        </button>
      </form>
    </div>
  </div>
</div>
`;
